modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -462,18 +487,68 @@ object ElasticRequest {
     }
   }
 
-  private def getActionAndMeta(requestType: String, parameters: List[(String, Any)]): String =
-    parameters.collect { case (name, Some(value)) => s""""$name" : "${value.toString}"""" }
-      .mkString(s"""{ "$requestType" : { """, ", ", " } }")
+  sealed trait UpdateRequest
+      extends ElasticRequest[UpdateOutcome]
+      with HasRefresh[UpdateRequest]
+      with HasRouting[UpdateRequest] {
 
-}
+    def docAsUpsert(value: Boolean): UpdateRequest
+
+    def docAsUpsertFalse: UpdateRequest
+
+    def docAsUpsertTrue: UpdateRequest
+
+    def upsert[A: Schema](doc: A): UpdateRequest
+  }
 
-sealed abstract class CreationOutcome
+  private[elasticsearch] final case class Update(
+    index: IndexName,
+    id: DocumentId,
+    doc: Option[Document],
+    docAsUpsert: Option[Boolean],
+    refresh: Option[Boolean],
+    routing: Option[Routing],
+    script: Option[Script],
+    upsert: Option[Document]
+  ) extends UpdateRequest { self =>
+    def docAsUpsert(value: Boolean): UpdateRequest =
+      self.copy(docAsUpsert = Some(value))
 
-case object AlreadyExists extends CreationOutcome
-case object Created       extends CreationOutcome
+    def docAsUpsertFalse: UpdateRequest =
```

Comment:
Let's define `docAsUpsertFalse` and `docAsUpsertTrue` in the trait.

Reply:
It is already defined.